module/zfs/dmu_send.c

```diff
@@ -3772,7 +3857,12 @@ dmu_recv_stream(dmu_recv_cookie_t *drc, vnode_t *vp, offset_t *voffp,
 		int next_err = 0;
 
 		while (next_err == 0) {
-			free_err = dmu_free_long_object(rwa->os, obj);
+			if (drc->drc_raw) {
+				free_err = dmu_free_long_object(rwa->os, obj);
+			} else {
+				free_err = dmu_free_long_object_raw(rwa->os,
+				    obj);
```

Comment:
is this right?  we use dmu_free_long_object_raw() if it is *not* a raw stream?

Reply:
good catch will fix.